File: src/main/java/com/topface/topface/requests/DeleteBookmarksRequest.java
package com.topface.topface.requests;

import android.content.Context;

import com.topface.topface.requests.handlers.BlackListAndBookmarkHandler;

import java.util.ArrayList;
import java.util.List;

public class DeleteBookmarksRequest extends DeleteAbstractUsersRequest {
    public static String SERVICE_NAME = "bookmark.delete";

    public DeleteBookmarksRequest(List<String> userIds, Context context) {
        super(userIds, context);
        ArrayList<Integer> ids = new ArrayList<>();
        for (int i = 0; i < userIds.size(); i++) {
            ids.add(Integer.parseInt(userIds.get(i)));
        }
        callback(new BlackListAndBookmarkHandler(getContext(), BlackListAndBookmarkHandler.ActionTypes.BOOKMARK, ids, false));
    }

    public DeleteBookmarksRequest(int id, Context context) {
        super(Integer.toString(id), context);
    }

    @Override
    public String getServiceName() {
        return SERVICE_NAME;
    }
}

File: src/main/java/com/topface/topface/api/requests/DeleteBookmarksRequest.kt
package com.topface.topface.api.requests

import com.google.gson.JsonObject
import com.topface.topface.api.responses.Completed
import com.topface.topface.api.stringListToJsonArray
import java.util.*

class DeleteBookmarksRequest(val userIds: ArrayList<String>) : BaseScruffyRequest<Completed>() {

    override fun getMethod() = "bookmark.delete"

    override fun createJson(json: JsonObject) = with(json) {
        val ids = userIds.map {
            it.substringAfter(":", it)
        }.stringListToJsonArray()
        add("userIds", ids)
    }

    override fun getResponseClass(): Class<Completed> = Completed::class.java
}